### PR TITLE
Luke's Proposed Changes

### PR DESCRIPTION
**Pull request by lukeeey**
### Brief
These are my proposed changes to the API of Cloudburst. 
They mostly consist of small things as well as extra Javadoc comments on existing methods.

I'm trying to keep the changes relatively small for now for simplicity.

**I don't expect this PR to be merged, and I'm still going to add to it regardless**

### Thoughts
You will find comments in the code prefixed with `LUKES NOTE:`. My thoughts that can't be conveyed in the code of this PR are as follows:

* `server` field among others should be declared `transient` in the Entity/Player/whatever classes in Cloudburst Server. This would help if the classes are serialized with Gson and such, or if you want to implement `Serializable`.
* The `util` package should really be cleaned up. Yes, I know its for misc utilities but we can do better.
* It would be nice to separate server property related methods from the `Server` class, although this just means extra code for plugins. Maybe not.

### CODE REVIEW

File: src/main/java/org/cloudburstmc/api/player/Player.java [hunk 1]
```diff
@@ -126,12 +127,6 @@
 
     Level getLevel();
 
-    boolean isSpectator();
```
Comment (lukeeey) [inline src/main/java/org/cloudburstmc/api/player/Player.java:129]: These are pretty useless methods and are just lazy, so I removed them.

File: src/main/java/org/cloudburstmc/api/util/TextFormat.java [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+     *
+     * @param input String to clean.
+     * @return A copy of the input string, without any formatting.
+     */
+    public static String clean(final String input) {
+        return clean(input, false);
+    }
+
+    /**
+     * Cleans the given message of all format codes.
```
Comment (lukeeey) [inline src/main/java/org/cloudburstmc/api/util/TextFormat.java:175]: Copied over the javadoc comment from above and modified it as this method did not have one.

File: src/main/java/org/cloudburstmc/api/util/TextFormat.java [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+    RESET('r', 0x15);
+
+    /**
+     * The special character which prefixes all format codes. Use this if
+     * you need to dynamically convert format codes from your custom format.
+     */
+    public static final char ESCAPE = '\u00A7';
+
+    private static final Pattern CLEAN_PATTERN = Pattern.compile("(?i)" + ESCAPE + "[0-9A-GK-OR]");
+    private final static Map<Integer, TextFormat> BY_ID = new TreeMap<>();
```
Comment (lukeeey) [inline src/main/java/org/cloudburstmc/api/util/TextFormat.java:112]: I'm not sure why this is a `TreeMap` when the below field is `HashMap`, but I removed the usage of guava's `Maps.newTreeMap()` here anyway.

File: src/main/java/org/cloudburstmc/api/plugin/annotations/Plugin.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * Defines a plugin.
+ *
+ * The main class of every plugin must contain this annotation at the class
+ * level so as to identify the plugin.
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Plugin {
```
Comment (lukeeey) [inline src/main/java/org/cloudburstmc/api/plugin/annotations/Plugin.java:16]: Just added Javadoc methods to this class and moved the location.

File: src/main/java/org/cloudburstmc/api/plugin/annotations/Dependency.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Defines a plugin dependency.
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Dependency {
```
Comment (lukeeey) [inline src/main/java/org/cloudburstmc/api/plugin/annotations/Dependency.java:13]: Just added Javadoc methods to this class and moved the location.

File: src/main/java/org/cloudburstmc/api/Server.java [hunk 6]
```diff
@@ -20,32 +19,96 @@
...
+     */
+    Collection<? extends Player> getOnlinePlayers();
+
+    /**
+     * Find a player by their username.
+     *
+     * @param name the name of the player
+     * @return an instance of that player if online
+     */
+    Player findPlayer(String name);
```
Comment (lukeeey) [inline src/main/java/org/cloudburstmc/api/Server.java:63]: I changed this to `findPlayer` as `find` better conveys the possibility of not finding a player (if they aren't online)

File: src/main/java/org/cloudburstmc/api/Server.java [hunk 7]
```diff
@@ -20,32 +19,96 @@
...
     String BROADCAST_CHANNEL_USERS = "cloudburst.broadcast.user";
 
+    /**
+     * Gets the name of this server implementation.
+     *
+     * @return name of this server
+     */
     String getName();
 
-    String getVersion();
```
Comment (lukeeey) [inline src/main/java/org/cloudburstmc/api/Server.java:25]: Removed this and replaced it with `getMinecraftVersion` as this is much clearer

File: src/main/java/org/cloudburstmc/api/event/inventory/CraftItemEvent.java [hunk 8]
```diff
@@ -31,7 +31,7 @@ public CraftItemEvent(Player source, List<ItemStack> input, Recipe recipe) {
 
     public CraftItemEvent(Player player, ItemStack[] input, Recipe recipe) {
         this.player = player;
-        this.input = Lists.newArrayList(input);
+        this.input = new ArrayList<>();
```
Comment (Extollite) [inline src/main/java/org/cloudburstmc/api/event/inventory/CraftItemEvent.java:34]: `new ArrayList<>(input)`?
Also I wonder, do we want input list to be mutable?
I feel like we don't, so `getInput` should return copy of input list.

**Pull request closed without merging**